packages/thirdweb/src/wallets/in-app/web/lib/in-app-account.ts

```diff
@@ -238,13 +239,22 @@ export class IFrameWallet {
         };
       },
       async signMessage({ message }) {
-        const messageDecoded =
-          typeof message === "string" ? message : message.raw;
+        // in-app wallets use ethers to sign messages, which always expects a string (or bytes maybe but string is safest)
+        const messageDecoded = (() => {
+          if (typeof message === "string") {
+            return message;
+          }
+          if (message.raw instanceof Uint8Array) {
+            return uint8ArrayToString(message.raw);
```

Comment:
shouldn't this be `uint8ArrayToHex` ?

Reply:
Hex is what breaks in-app wallets, ethers just assumes they're a string and signs them as one (which still works in most cases but not with SIWE)